File: benchmarks/bench_15.py
import time
import sys
sys.setrecursionlimit(200000)

def is_prime(n):
    if n < 2:
        return 0
    i = 2
    while i * i <= n:
        div = n // i
        rem = n - (div * i)
        if rem == 0:
            return 0
        i += 1
    return 1

def prime_count(limit):
    c = 0
    i = 2
    while i < limit:
        if is_prime(i) == 1:
            c += 1
        i += 1
    return c


t1 = time.perf_counter()
res = prime_count(10000)
t2 = time.perf_counter()
print(res)
print(f"{t2 - t1:.6f}")

File: benchmarks/bench_15.go
package main

import (
	"fmt"
	"time"
)

func is_prime(n int64) int64 {
    if n < 2 {
        return 0
    }
    i := int64(2)
    for i*i <= n {
        div := n / i
        rem := n - (div * i)
        if rem == 0 {
            return 0
        }
        i += 1
    }
    return 1
}
func prime_count(limit int64) int64 {
    c := int64(0)
    i := int64(2)
    for i < limit {
        if is_prime(i) == 1 {
            c += 1
        }
        i += 1
    }
    return c
}

func main() {
    t1 := time.Now()
    res := prime_count(10000)
    t2 := time.Now()
    fmt.Println(res)
    fmt.Printf("%.6f\n", t2.Sub(t1).Seconds())
}
